Solaris 64-bits defines a 64-bits off_t, and hence doesn't need
_LARGE_FILES or something similar; the env is fully 64-bits, of course.
In that case, however, ac_cv_sys_large_files is unset, leading this
function to believe that the system defaults to 32-bits off_t, which is
obviously wrong, and leads to linkage failures lateron due to missing
xxx64 functions.

--- m4/ac_sys_largefile_sensitive.m4
+++ m4/ac_sys_largefile_sensitive.m4
@@ -38,7 +38,7 @@
 # we know about some internals of ac_sys_largefile here...
 AC_MSG_CHECKING(whether system differentiates 64bit off_t by defines)
 ac_cv_sys_largefile_sensitive="no"
-if test ".$ac_cv_sys_file_offset_bits$ac_cv_sys_large_files" != ".nono"
+if test ".$ac_cv_sys_file_offset_bits${ac_cv_sys_large_files-no}" != ".nono"
 then ac_cv_sys_largefile_sensitive="yes"
   AC_DEFINE(LARGEFILE_SENSITIVE, 1,
   [whether the system defaults to 32bit off_t but can do 64bit when requested])
